core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -420,6 +420,21 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val STORAGE_SHUFFLE_DECOMMISSION_ENABLED =
+    ConfigBuilder("spark.storage.decommission.shuffle_blocks")
+      .doc("Whether to transfer shuffle blocks during block manager decommissioning. Requires " +
+        "an indexed shuffle resolver (like sort based shuffe)")
```

Comment:
It would be great if we have a defensive code somewhere in this PR for this requirement. For example, `assert()` in the code?
> Requires an indexed shuffle resolver (like sort based shuffe)

Reply:
Oh actually should change this wording, it can work a little more generally now. Any implementation which implements a given trait. I'm not sure about adding an assert here, we do log it when we first try and use it though.

Reply:
Got it~